src/Raven.Server/Documents/Handlers/DocumentReplicationHandler.cs

```diff
@@ -1,16 +1,190 @@
     {
         [RavenAction("/databases/*/replication/topology", "GET")]
         public Task GetReplicationTopology()
         {
             HttpContext.Response.StatusCode = 404;
             return Task.CompletedTask;
         }
+
+        [RavenAction("/databases/*/replication/active-connections", "GET")]
+        public Task GetReplicationActiveConnections()
+        {
+            HttpContext.Response.StatusCode = (int)HttpStatusCode.OK;
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var incoming = new DynamicJsonArray();
+                foreach (var item in Database.DocumentReplicationLoader.IncomingConnections)
+                {
+                    incoming.Add(new DynamicJsonValue
+                    {
+                        ["SourceDatabaseId"] = item.SourceDatabaseId,
+                        ["SourceDatabaseName"] = item.SourceDatabaseName,
+                        ["SourceMachineName"] = item.SourceMachineName,
+                        ["SourceUrl"] = item.SourceUrl
+                    });
+                }
+
+                var outgoing = new DynamicJsonArray();
+                foreach (var item in Database.DocumentReplicationLoader.OutgoingConnections)
+                {
+                    outgoing.Add(new DynamicJsonValue
+                    {
+                        ["Url"] = item.Url,
+                        ["Database"] = item.Database,
+                        ["Disabled"] = item.Disabled,
+                        ["IgnoredClient"] = item.IgnoredClient,
+                        ["SkipIndexReplication"] = item.SkipIndexReplication,
+                        ["SpecifiedCollections"] = item.SpecifiedCollections
+                    });
+                }
+
+                context.Write(writer, new DynamicJsonValue
+                {
+                    ["IncomingConnections"] = incoming,
+                    ["OutgoingConnections"] = outgoing
+                });
+            }
+            return Task.CompletedTask;
+        }
+
+        [RavenAction("/databases/*/replication/debug/outgoing-failures", "GET")]
+        public Task GetReplicationOugoingFailureStats()
+        {
+            HttpContext.Response.StatusCode = (int)HttpStatusCode.OK;
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var data = new DynamicJsonArray();
+                foreach (var item in Database.DocumentReplicationLoader.OutgoingFailureInfo)
+                {
+                    data.Add(new DynamicJsonValue
+                    {
+                        ["Key"] = new DynamicJsonValue
```

Comment:
Why are you using `Key`/`Value` as the property names?
Also, where are the actual _errors_ ? You let us know that this is failing, but not why